packages/components/src/suggestions/item.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import type { ForwardRefRenderFunction, FocusEvent, MouseEvent } from 'react';
+import classNames from 'classnames';
+import { escapeRegExp } from 'lodash';
+
+function escapeRegExpWithSpace( str: string ) {
+	return escapeRegExp( str ).replace( /\s/g, '\\s' );
+}
+
+const createTextWithHighlight = ( text: string, query: string ) => {
+	const re = new RegExp( '(' + escapeRegExpWithSpace( query ) + ')', 'gi' );
+	const parts = text.split( re );
+
+	// Replaces char code 160 (&nbsp;) with 32 (space)
+	const match = query.toLowerCase().replace( /\s/g, ' ' );
+
+	return parts.map( ( part, i ) => {
+		const key = text + i;
+		const lowercasePart = part.toLowerCase();
+		const spanClass = classNames( 'suggestions__label', {
+			'is-emphasized': lowercasePart === match,
+		} );
+
+		return (
+			<span key={ key } className={ spanClass }>
+				{ part }
+			</span>
+		);
+	} );
+};
+
+type Props = {
+	label: string;
+	hasHighlight: boolean;
+	query: string;
+	onMount: () => void;
+	onMouseDown: () => void;
+	onMouseOver: () => void;
+};
+
+const InnerItem: ForwardRefRenderFunction< HTMLButtonElement, Props > = (
+	{ label, hasHighlight, query, onMount, onMouseDown, onMouseOver },
+	forwardedRef
+) => {
+	React.useEffect( () => {
+		onMount();
+	}, [ onMount ] );
+
+	const handleMouseDown = React.useCallback(
+		( event: MouseEvent | FocusEvent ) => {
+			event.stopPropagation();
+			event.preventDefault();
+			onMouseDown();
+		},
+		[ onMouseDown ]
+	);
+
+	const handleMouseOver = React.useCallback( () => {
+		onMouseOver();
+	}, [ onMouseOver ] );
+
+	const className = classNames( 'suggestions__item', { 'has-highlight': hasHighlight } );
+
+	return (
+		<button
+			className={ className }
+			onMouseDown={ handleMouseDown }
+			onFocus={ handleMouseDown }
+			onMouseOver={ handleMouseOver }
+			ref={ forwardedRef }
+		>
+			{ createTextWithHighlight( label, query ) }
+		</button>
+	);
+};
+
+const Item = React.forwardRef( InnerItem );
+
+Item.defaultProps = {
```

Comment:
Does it make sense to add these as default object properties (`{ label, hasHigllight = false, ... }`) now that we're using a functional component?

Reply:
Sure, if that's preferred.